src/bp/ui-admin/src/Pages/Bot/Details.jsx

```diff
@@ -238,6 +280,8 @@ class Bots extends Component {
       this.setState({ error: null })
     }
 
+    // [TODO] max.cloutier 2020.01.17 Add indications that this will submit the form OR change the behavior so it saves only on save btn click
+    // If it can't be uploaded without saving, it should probably be a dedicated option that isn't part of this form
```

Comment:
The image gets uploaded, however it's not active until the form is saved (it uploads the media, then update the url in the state, which is saved to the bot when the button Save is pressed). It's true that it doesn't get deleted if you press cancel, but the issue is more with the lack of management for media (eg: no delete route / media list) than an issue with the bot details page.

Also, if you want to add 2 pictures, you need to open the form twice, which gives a weird experience. I think it would make more sense for now to keep the initial flow (upload the images and set the url in the state, then save the bot config when the button is pressed). 

The experience will already be better with the cancel button prompting when something was changed

Reply:
@allardy the save behavior right after the upload was already in place when I implemented my changes, that's why I didn't touch it as I wasn't sure if it would affect something I didn't know about.

I would suggest the following:
1. Upload occurs
2. On upload complete, a message appears on the relevant image saying that the save button must be pressed in order to save the change to the image
3. Cancel button shows the confirm message if there was an upload saying that there are unsaved changes.

Reply:
Yes, that would be awesome, and it would fix all the UX issues! You are absolutely right about the behavior, It was less visible since the bot config page stayed open!